From 677495f54c57c9b0f46ec983621b82392ddf92a0 Mon Sep 17 00:00:00 2001
From: Wu Liang feng <wulf@rock-chips.com>
Date: Mon, 7 Dec 2015 14:23:51 +0800
Subject: [PATCH] USB: dwc_otg_310: pcd: fix duplicate send EP0 IN ZLP

From v3.00a dwc2 controller add a new interrupt 'StsPhseRcvd'
(Status Phase Received For Control Write) in DOEPINT. Refer
to dwc_otg_databook_v3.10, this interrupt is valid only for
control OUT endpoints, it indicates to the application that
the host has switched from data phase to the status phase
of a control write transfer. The application can use this
interrupt to ACK or STALL the Status phase.
We use v3.10a dwc2 on rockchip soc, but actually we don't
need to send Zero-Length Packet for the IN status phase
in StsPhseRcvd int handler in DMA mode. The handle_ep0->
ep0_complete_request will call do_setup_in_status_phase to
send EP0 IN ZLP.
TEST=set gadget work as usb rndis and connect to ubuntu.

Change-Id: I5203248a687ddbac097c74bb0f856385776df6d9
Signed-off-by: Wu Liang feng <wulf@rock-chips.com>
---
 drivers/usb/dwc_otg_310/dwc_otg_pcd_intr.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/usb/dwc_otg_310/dwc_otg_pcd_intr.c b/drivers/usb/dwc_otg_310/dwc_otg_pcd_intr.c
index 29c054e2b879..46ba82353633 100644
--- a/drivers/usb/dwc_otg_310/dwc_otg_pcd_intr.c
+++ b/drivers/usb/dwc_otg_310/dwc_otg_pcd_intr.c
@@ -4908,12 +4908,10 @@ do { \
 				deptsiz.d32 =
 				    DWC_READ_REG32(&core_if->dev_if->
 						   out_ep_regs[0]->doeptsiz);
-				if ((core_if->dma_desc_enable)
-				    || (core_if->dma_enable
-					&& core_if->snpsid >=
-					OTG_CORE_REV_3_00a)) {
+				if (core_if->dma_desc_enable) {
 					do_setup_in_status_phase(pcd);
 				}
+
 			}
 
 			/* Endpoint disable      */
-- 
2.35.3

